acceptance/mockAgent/daemon.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2015 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"fmt"
+	"net"
+	"net/rpc"
+	"net/rpc/jsonrpc"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/control-center/serviced/domain/host"
+	"github.com/control-center/serviced/rpc/agent"
+	"github.com/control-center/serviced/utils"
+
+	"github.com/zenoss/glog"
+)
+
+type daemon struct {
+	hostConfig *HostConfig
+	rpcServer  *rpc.Server
+	host       *host.Host
+}
+
+func newDaemon(hostConfig *HostConfig, rpcServer *rpc.Server) (*daemon, error) {
+	d := &daemon{
+		hostConfig: hostConfig,
+		rpcServer:  rpcServer,
+	}
+	return d, nil
+}
+
+func (d *daemon) run() (err error) {
+	if err := d.hostConfig.setDefaults(); err != nil {
+		return fmt.Errorf("unable to initialize host configuration: %v", err)
+	}
+
+	if err := d.buildHost(); err != nil {
+		return fmt.Errorf("failed to build Host domain object: %v", err)
+	}
+
+	d.startRPC()
+	if err := d.registerAgentRPC(); err != nil {
+		glog.Fatal(err)
+	}
+
+	signalC := make(chan os.Signal, 10)
+	signal.Notify(signalC, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP)
+	sig := <-signalC
+	glog.Infof("Shutting down due to interrupt - %v", sig)
+	return nil
+}
+
+func (d *daemon) startRPC() {
+	listener, err := net.Listen("tcp", d.hostConfig.Listen)
+	if err != nil {
+		glog.Fatalf("Unable to bind to port %s. Is another instance running?", d.hostConfig.Listen)
+	}
+
+	d.rpcServer.HandleHTTP(rpc.DefaultRPCPath, rpc.DefaultDebugPath)
+
+	glog.Infof("Listening on %s", listener.Addr().String())
+	go func() {
+		for {
+			glog.Infof("listening:Accept()")
+			conn, err := listener.Accept()
+			if err != nil {
+				glog.Fatalf("Error accepting connections: %s", err)
+			}
+			glog.Infof("listening:ServeCodec()")
+			go d.rpcServer.ServeCodec(jsonrpc.NewServerCodec(conn))
+			// err = d.rpcServer.ServeRequest(jsonrpc.NewServerCodec(conn))
+			// if err != nil {
+			// 	glog.Fatalf("Error serving request: %s", err)
+			// }
+		}
+	}()
+}
+
+type MockServer struct {
```

Comment:
Please move MockServer and all of it's methods to a new file, mockAgent.go and rename the class to MockAgent